pkg/didcomm/common/service/service.go

```diff
@@ -16,7 +16,7 @@ import (
 // Handler provides protocol service handle api.
 type Handler interface {
 	// HandleInbound handles inbound messages.
-	HandleInbound(msg *DIDCommMsg) (string, error)
+	HandleInbound(msg *DIDCommMsg, myDID string, theirDID string) (string, error)
```

Comment:
It looks like my myDID is redundant. We can get my myDID using a  `GetDID` function. e.g 
`myDID := GetDID(theirDID)`. According to that line https://github.com/hyperledger/aries-framework-go/pull/938/files#diff-b429c177e8c9b423d01fcc0d7d2383e9R81

Reply:
This shouldn’t be guaranteed. An agent could have multiple connections to the same “theirDIDs” with different “myDIDs”.

Reply:
Yeah, but it is a map. I mean only two DIDs can be related to each other. According to code we have. See link ^

Reply:
Otherwise GetDID(theirDID) should return a list of myDids. Which means you can use any of them. Right?

Reply:
I am a bit confused. Why do we need multiple myDIDs by on theirDid ? If connection is established doesn’t make sense to create one more.

Reply:
It's useful for communicating with a public DID - you might want multiple DIDs you can use to communicate with them, for privacy.

Reply:
A Lookup function is needed in the didconnection package to make this more useful. See prior comment.

Reply:
Since myDID and theirDID are parameters to HandleInbound, we don't need the theirDID -> myDID map. As parameters to HandleInbound, the DIDs can now be any values for any connection, even if theirDID is used in multiple connections.